app/validate_txs.go

```diff
@@ -8,11 +8,44 @@ import (
 	"github.com/cosmos/cosmos-sdk/telemetry"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
+	square "github.com/celestiaorg/go-square/v2"
 	"github.com/celestiaorg/go-square/v2/tx"
 
 	"github.com/celestiaorg/celestia-app/v4/pkg/appconsts"
 )
 
+// FilteredBuilder filters txs and blobs using a copy of the state and tx validity
+// rules before adding it the square.
+type FilteredBuilder struct {
```

Comment:
maybe we could call this `FilteredSquareBuilder` to be more specific about what the type is used for, I think it would help the reader

I would also suggest moving the code in this file (`validate_txs.go` -> `square.go` or something similar). I think `square_size.go` exists in this pkg with a single small func which could maybe be moved and live next to all things square 🟦 (although, I think its a method on the app)